### ISSUE TITLE
Wrong result for a simple expression

### ISSUE THREAD

**Issue by tomlux** [vatbub/mathevaluator#3]
Hy,
I'm making this expression:
 3.93 + 2.87 + 2.23 + 2.51 + 2.05 + 2+2.65 + 2.66 + 2.6 + 2.6 + 2.6 + 2.6 + 1.9 + 2.6 + 2.6

The result shoud be: **38.4**

But I'm getting when running with the MathExpression as result: **38.400000000000006**

Here the simple code:
`        MathExpression expression = new MathExpression("3.93+2.87+2.23+2.51+2.05+2+2.65+2.66+2.6+2.6+2.6+2.6+1.9+2.6+2.6");
        System.out.println("expression.value = " + expression.evaluate().getValue());
`

Is this a bug in your library or what is here wrong?

Thanks,
 Tom

**Comment by vatbub** [comment]
Hey Tom, the library saves number literals as `double`s, the calculations are hence carried out using double precision. To me, this looks like usual rounding errors which occur when doing floating point calculations. I will have a closer look at it later today.

**Comment by vatbub** [comment]
I just tried your example and modified it a little:

```java
MathExpression expression = new MathExpression("3.93+2.87+2.23+2.51+2.05+2+2.65+2.66+2.6+2.6+2.6+2.6+1.9+2.6+2.6");
double expected = 3.93 + 2.87 + 2.23 + 2.51 + 2.05 + 2 + 2.65 + 2.66 + 2.6 + 2.6 + 2.6 + 2.6 + 1.9 + 2.6 + 2.6;
Number result = expression.evaluate();
System.out.println("expression.value = " + result.getValue());
System.out.println("expected result: = " + expected);
```

I got the following console output:

```
expression.value = 38.400000000000006
expected result: = 38.400000000000006
```

This shows again that the issue is rounding related to floating point rounding errors. If you wish to read more about why this happens, you can read [this overview on StackExchange](https://softwareengineering.stackexchange.com/questions/101163/what-causes-floating-point-rounding-errors) or [this paper](https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html) if you wish to get more in-depth information.

Let me know if you need any other help with the library :)

**Comment by vatbub** [comment]
Apart from that, you may want to pull and run `mvn install` again as I just fixed issues #1 and #2 .
Cheers :)
